shacl12-core/index.html

```diff
@@ -1702,7 +1703,7 @@ <h3>Node Expressions</h3>
 					The evaluation may also result in an <dfn>evaluation failure</dfn>.
```

Comment:
`eval(expr, activeGraph, scope)`

Does scope include the parameters named by IRI?
scope is SPARQL variable name to node (and used as "focusNode")

2:: What about postional arguments `[ fn;strlen ( someWayToGetInputFocuNode   2   3 ) ]`?

Reply:
1) No, the definition of the eval function (or process or algorithm) would include details on how to fetch the parameter values from the given expr node. In some cases these are further node expressions that require nested evaluation. But the scope is a separate concept that gets passed into the execution engine from the outside.

2) I hope we can move the details of the syntax into SHACL 1.2 Node Expr. Core doesn't yet need this and it will make more sense when there are real functions in the library (for example etc). Alternatively, we could cover some of it in SHACL-SPARQL, in the part about sh:Functions.

(For housekeeping I'll mark this resolved but please reopen if this is not good enough).

Reply:
SHACL Core needs to define node expression evaluation if it is to be the basis for other documents. A partial just means the NE document is actually starting point and SHACL-SPARQL is based on SHACL-NE.

In WG discussions, parameter lists have been common.

(And then the doc can have another example that is not using `sh:select` - `sh:targetNode` would seem like a good one.)